eng/targets/Wix.Common.props

```diff
@@ -3,8 +3,8 @@
 
   <PropertyGroup>
     <SchemaVersion>2.0</SchemaVersion>
-    <ProductVersion>3.11</ProductVersion>
-    <WixVersion>3.11.1</WixVersion>
+    <ProductVersion>3.14</ProductVersion>
+    <WixVersion>3.14.0-dotnet</WixVersion>
```

Comment:
As @joeloff said in #18091,
> we should give it a try

There's a very good chance this package will _not_ change whether the wixstdba.dll file is signed. This package is an old fork and not that far removed from 3.11.1. Don't make this change if it leaves the file unsigned.